onnxruntime/core/session/inference_session.cc

```diff
@@ -422,6 +422,16 @@ common::Status InferenceSession::Initialize() {
                                                    std::make_unique<CPUExecutionProvider>(epi)));
     }
 
+    const auto first_provider = std::cbegin(execution_providers_);
+    if (!session_options_.enable_sequential_execution &&
+        first_provider->get()->Type() != onnxruntime::kCpuExecutionProvider) {
+      LOGS(*session_logger_, ERROR) << "Parallel execution is only supported "
+                                       "for the CPU Execution Provider currently.";
+      return common::Status(common::ONNXRUNTIME, common::FAIL,
```

Comment:
one minor nit: is that the status code is not FAIL really. It should be INVALID_ARGUMENT. don't have to fix in this PR.

Reply:
That's true. I resolved it (as the CIs have not made much of a start anyway).